Allow ESM module for custom middlewares

5ika opened the pull request:
### What does it do?

Fix issue #14146 

### Why is it needed?

With TS version of Strapi, we can use ESM module imports but we can't import custom middleware in `.ts` module file.

### How to test it?

See https://github.com/strapi/strapi/issues/14146

### Related issue(s)/PR(s)

#14146

strapi-cla commented:
[![CLA assistant check](https://cla.strapi.io/pull/badge/not_signed)](https://cla.strapi.io/strapi/strapi?pullRequest=14280) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.strapi.io/strapi/strapi?pullRequest=14280) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.strapi.io/check/strapi/strapi?pullRequest=14280) it.</sub>

Convly reviewed (changes requested):
Hey, thank you for taking the time to open a PR,

If we want to enable loading ESM here, we could use the `import-default` util from [here](https://github.com/strapi/strapi/blob/features/api-token-v2/packages/core/strapi/lib/utils/import-default.js#L1) and move it to the `@strapi/utils` package, so that we're consistent with the rest of the logic where we allow using ESMs.

5ika commented:
@Convly Thanks ! I've updated the code to use import-default function.
I'm not sure I understand the last part of your comment: you mean move the function to @strapi/utils then update all occurrences in the code base ?

Convly commented:
> @Convly Thanks ! I've updated the code to use import-default function. I'm not sure I understand the last part of your comment: you mean move the function to @strapi/utils then update all occurrences in the code base ?

Hey, really sorry for the late answer, it got lost in the Github notifications.
That's I was suggesting indeed, we could move the content of [this file](https://github.com/strapi/strapi/blob/main/packages/core/strapi/lib/utils/import-default.js) to the [strapi/utils](https://github.com/strapi/strapi/tree/main/packages/core/utils/lib) package.

5ika commented:
> Hey, really sorry for the late answer, it got lost in the Github notifications. That's what I was suggesting indeed, we could move the content of [this file](https://github.com/strapi/strapi/blob/main/packages/core/strapi/lib/utils/import-default.js) to the [strapi/utils](https://github.com/strapi/strapi/tree/main/packages/core/utils/lib) package.

@Convly No problem for the delay and thank you for your response. I've moved import-default to strapi/utils package as asked :+1:

Convly commented on packages/core/strapi/lib/core/app-configuration/load-config-file.js:
```diff
@@ -4,7 +4,7 @@ const path = require('path');
 const fs = require('fs');
 const { templateConfiguration, env } = require('@strapi/utils');
```
```suggestion
const { templateConfiguration, env, importDefault } = require('@strapi/utils');
```

Convly commented on packages/core/strapi/lib/services/server/middleware.js:
```diff
@@ -2,6 +2,7 @@
 
 const path = require('path');
 const { propOr, isArray, isNil } = require('lodash/fp');
+const importDefault = require('@strapi/utils/lib/import-default');
```
```suggestion
const { importDefault } = require('@strapi/utils');
```

Convly reviewed (changes requested):
Thanks, I'm afraid those files would break if we don't change the importDefault require
- /packages/core/strapi/lib/core/loaders/apis.js
- /packages/core/strapi/lib/core/loaders/middlewares.js
- /packages/core/strapi/lib/core/loaders/policies.js
- /packages/core/strapi/lib/core/loaders/src-index.js

5ika commented:
> Thanks, I'm afraid those files would break if we don't change the importDefault require
> 
>     * /packages/core/strapi/lib/core/loaders/apis.js
> 
>     * /packages/core/strapi/lib/core/loaders/middlewares.js
> 
>     * /packages/core/strapi/lib/core/loaders/policies.js
> 
>     * /packages/core/strapi/lib/core/loaders/src-index.js

Oh, sorry. I thought I've updated all occurrences. It's now updated.
Thank you

Convly reviewed (approved):
LGTM
Thanks for your contribution :rocket:

Convly commented:
_(I'll just do a last round of verification (probably today or tomorrow) before merging)_
